src/utils.jl

```diff
@@ -233,15 +233,56 @@ Returns a tuple of the parameters of the SimpleChain `sc`, as a view of the para
 """
 function params(sc::SimpleChain, p::AbstractVector, inputdim = nothing)
   @unpack layers = sc
-  A = _params(layers, pointer(p), chain_input_dims(sc, inputdim))
+  A = _walk_chain(Val{:param}(), layers, pointer(p), chain_input_dims(sc, inputdim))
   _add_memory(A, p)
 end
-_params(::Tuple{}, _, __) = ()
-function _params(layers, p, inputdim)
-  A, p, outputdim = _getparams(first(layers), p, inputdim)
-  B = _params(Base.tail(layers), p, outputdim)
+"""
+    weights(sc::SimpleChain, p::AbstractVector, inputdim = nothing)
+
+Returns a tuple of the weights (parameters other than biases) of the SimpleChain `sc`, as a view of the parameter vector `p`.
+"""
+function weights(sc::SimpleChain, p::AbstractVector, inputdim = nothing)
+  @unpack layers = sc
+  A = _walk_chain(Val{:weight}(), layers, pointer(p), chain_input_dims(sc, inputdim))
+  _add_memory(A, p)
+end
+"""
+    biases(sc::SimpleChain, p::AbstractVector, inputdim = nothing)
+
+Returns a tuple of the biases of the SimpleChain `sc`, as a view of the parameter vector `p`.
+"""
+function biases(sc::SimpleChain, p::AbstractVector, inputdim = nothing)
+  @unpack layers = sc
+  A = _walk_chain(Val{:bias}(), layers, pointer(p), chain_input_dims(sc, inputdim))
+  _add_memory(A, p)
+end
+
+# definitions that happen to be right in most cases to save up
+# from implementing too much
+_get(::Val{:param}, x) = x
+_get(::Val{:weight}, ::Nothing) = nothing
+_get(::Val{:weight}, x) where {A,B} = x
+_get(::Val{:weight}, x::Tuple{A,B}) where {A,B} = first(x)
+_get(::Val{:bias}, ::Nothing) = nothing
+_get(::Val{:bias}, x) where {A,B} = nothing
+_get(::Val{:bias}, x::Tuple{A,B}) where {A,B} = last(x)
+@inline function _getparams(f::F, layer, p, inputdim) where {F}
```

Comment:
```suggestion
@inline function _getparams(f, layer, p, inputdim)
```
`F` is not used - is the `where {F}` superfluous?

Reply:
`F` is `::Val`, but originally I was thinking it could be a function.
If it's a function, it'd be needed to force specialization.

Reply:
Ok, I don't mind